sentinelsat/sentinel.py

```diff
@@ -764,20 +764,26 @@ def _download(self, url, path, session, file_size):
         else:
             already_downloaded_bytes = 0
         downloaded_bytes = 0
-        with closing(session.get(url, stream=True, auth=session.auth, headers=headers)) as r, \
-                closing(self._tqdm(desc="Downloading", total=file_size, unit="B",
-                                   unit_scale=True, initial=already_downloaded_bytes)) as progress:
-            _check_scihub_response(r, test_json=False)
-            chunk_size = 2 ** 20  # download in 1 MB chunks
-            mode = 'ab' if continuing else 'wb'
-            with open(path, mode) as f:
-                for chunk in r.iter_content(chunk_size=chunk_size):
-                    if chunk:  # filter out keep-alive new chunks
-                        f.write(chunk)
-                        progress.update(len(chunk))
-                        downloaded_bytes += len(chunk)
-            # Return the number of bytes downloaded
-            return downloaded_bytes
+        while True:
```

Comment:
* avoid the `while True` construction, as this will retry indefinetly
* we have a retry logic in place in the `download_all` function with `max_attempts`: https://github.com/sentinelsat/sentinelsat/blob/master/sentinelsat/sentinel.py#L500
* `timeout` should be a parameter of `download_all`, just like `max_attempts`, with a default value

This construction would break the retry logic.